### ISSUE TITLE
「加工禁止文字」が投稿に含まれていたとき、投稿前に介入する措置

### ISSUE THREAD

**Issue by KisaragiEffective** [niwaniwa/misskey#136]
### 概要

企業のロゴやオリジナルキャラクター(以下、まとめて保護されたアセット)というのは本来様々な知的財産権に守られていて連合に乗せたり公衆送信可能化することは法的なリスクが避けられない側面があります。また、保護されたアセットはそれぞれに定められた一定のガイドライン(以下、アセットガイドライン)にそって利用が許諾されることがありますが、アスペクト比の変更、色相の変更、回転、指差し、などその禁止事項は多岐にわたります。
そこで、以下の変更を提案します。

1. 絵文字について、新しく`Unmodifiable`及び`Non-Redistributable`というフラグを立てられるようにする。デフォルトはどちらもfalse。
2. ユーザーが投稿する際、`Unmodifiable`がついた絵文字が直接または間接的にplainタグを除くMFMコンポーネントの引数として検出された際、当該の絵文字が使えない旨のエラーをバックエンドから表示し投稿を却下する。
3. ユーザーが投稿する際、`Unmodifiable`がついた絵文字が投稿に含まれ、かつ`scale`、`rotate`、`position`のタグが1つ以上含まれている時、当該の絵文字のアセットガイドラインに違反する可能性がある警告を表示する。
4. ユーザーが投稿する際、`Non-Redistributable`フラグがついた絵文字が検出された場合、その投稿の連合フラグを「連合しない」にする (注1)。
5. 通常通りの投稿前の処理に入る。

(注1): 混乱を招くと予想されるため、絵文字だけ連合しない、のほうが良いかもしれません

### 目的

以下の効果が期待できます。

1. ユーザーが知らず知らずのうちにアセットガイドラインに違反することを避ける
2. サーバーが再配布できない絵文字を再配布してしまうことによって著作権者から何らかの措置を取られるリスクを減らす

**Comment by Sayamame-beans** [comment]
Non-Redistributableに限っては既にあります (ローカル限定絵文字。にりみすだと私が入ってる絵文字はそれですね)

**Comment by anatawa12** [comment]
アセットガイドライン位各自でやってくれという気持ち。

**Comment by KisaragiEffective** [comment]
> アセットガイドライン位各自でやってくれという気持ち。

実際コストと対価が見合うかどうか？というのは散逸するLTLで検討するには不適当なのでGitHubにあげてみました。

**Comment by Sayamame-beans** [comment]
何というか、余分に煩雑になる気がしています。
運営として、危険と思うなら登録の時点で弾いておくか、或いは申し立てがあったら対応する形で行くかという判断をすると思いますし、"加工禁止"を課してまでカスタム絵文字として利用出来る必要があるか微妙な気もします。

…ふとした思い付きなのですが、本文中では使えない"リアクション専用"のカスタム絵文字とかどうでしょうか。
需要があるかは分かりませんが、あるとするならバランスは良いと思います?
